third-party/py/pex/pex/compiler.py

```diff
@@ -58,6 +60,17 @@ def main(root, relpaths):
 """
 
 
+@contextlib.contextmanager
+def named_temporary_file():
+    fp, path = tempfile.mkstemp()
+    os.close(fp)
+    try:
+        with open(path, 'w') as fp:
+            yield fp
+    finally:
+        os.remove(path)
```

Comment:
I hit that too, but it wasn't the subprocess call - it was the fp returned by mkstemp